pkg/controller/worker/actuator.go

```diff
@@ -28,4 +28,15 @@ type Actuator interface {
 	Reconcile(context.Context, *extensionsv1alpha1.Worker, *extensionscontroller.Cluster) error
 	// Delete deletes the Worker.
 	Delete(context.Context, *extensionsv1alpha1.Worker, *extensionscontroller.Cluster) error
+	// Restore read from worker.status.state field and deploy the machines and machineSet
+	Restore(context.Context, *extensionsv1alpha1.Worker, *extensionscontroller.Cluster) error
+	// Migrate delete the MCM, machineDeployments, mahicneClasees, machineClassSecrets,
+	// machineSets and the machines, then deletes the worker
```

Comment:
Maybe we should add additional information that actual VMs should not be deleted.